draft-birkholz-scitt-receipts.md

```diff
@@ -252,57 +251,49 @@ ProofElement = [
 ]
 
 LeafInfo = [
-    prefix: bstr,
-    type: LeafEntryType,
-    data: LeafReceiptData
+    writeset_hash: bstr,
+    commit_evidence: tstr,
+    sign_protected: empty_or_serialized_map
 ]
 ~~~
 
 ## Receipt Generation
 
-[TODO] this needs improvement
+The following steps must be followed to generate a receipt after the tree root has been signed:
 
 1. Let LEAF be the leaf in the Merkle tree for which a receipt should be generated.
 
-2. Let ROOT_HASH be the root hash of the Merkle tree that contains LEAF and SIGNATURE the signature over this root.
+2. Let ROOT_HASH be the root hash of the Merkle tree that contains LEAF, and SIGNATURE the signature over this root.
 
 3. Compute LEAF_HASH as the hash of LEAF.
 
 4. Generate an inclusion proof from LEAF_HASH to ROOT_HASH.
 
-5. Construct a LeafInfo structure with the implementation-specific prefix, the LeafEntryType, and the LeafReceiptData using the procedure defined by the leaf entry type.
+5. Construct a LeafInfo structure with the CCF writeset digest, the commit evidence, and the protected header parameters of the countersigner.
```

Comment:
Not being a native speaker, I have no opinion on the "write set" vs "writeset" distinction, but I think sticking to either "digest" or "hash" (see above) throughout would be more consistent.